### PR TITLE
Add Bonneville Power Administration

### PR DESCRIPTION
**Pull request by systemcatch**
Currently this is just the parser until we figure out how to display the area on the map.

ref #1040

### CODE REVIEW

**Comment by brunolajoie** [comment]
niiiice !

File: parsers/US_BPA.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    processed_data = []
+    for index, row in df.iterrows():
+        production = row.to_dict()
+
+        for key in keys_to_remove:
+            production.pop(key)
+
+        dt = production.pop('Date/Time')
+        mapped_production = {GENERATION_MAPPING[k]:v for k,v in production.items()}
```
Comment (maxbellec) [inline parsers/US_BPA.py]:
```python
mapped_production = {GENERATION_MAPPING[k]: v for k, v in production.items()
                     if k not in keys_to_remove}
```
might be slightly more elegant

**Comment by systemcatch** [comment]
The parser is all good to go I think, does anyone have thoughts on how to represent this area on the map?

**Comment by systemcatch** [comment]
Chatting to @alixunderplatz on slack we came to the conclusion that using the state of Washington as a rough representation is a good start.

Interestingly WattTime (developer of pyiso) have a map of the USA showing all of the balancing authority areas, see http://explorer.watttime.org/. They are using Leaflet to overlay these complex shapes onto a Mapbox base, @corradio could we do something similar?

**Comment by corradio** [comment]
If we have the shapes defined in lonlat space we can integrate them on the map, but else it will be messy.. I've looked in the past at using the WattTime geometries to use in ElectricityMap, but that turned out not to be a good solution as the zoom features will become imprecise.
Issue #143 might have the geometries already defined? Else let's use Washington?

Oli

**Comment by systemcatch** [comment]
Implemented the area.

us-bpa

**Comment by corradio** [comment]
woop woop!

**Comment by brunolajoie** [comment]
@systemcatch is on a killing spree! 💪

**Comment by jarek** [comment]
I know the U.S. power systems situation is a royal pain, but 1.3 GW nameplate of semi-lignite at [Centralia](https://en.wikipedia.org/wiki/Centralia_Power_Plant) just got coloured hydro-green ¯\_(ツ)_/¯

**Comment by systemcatch** [comment]
:scream: we should fix the US map areas asap.

**Comment by brunolajoie** [comment]
Daaaaaaaaaamn 🙀 🙀 🙀 🙀

**Pull request merged**